src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (typeof num === 'number') {
+      sum += num;
+
+      return adder;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
```

Comment:
The `adder` function should not take any arguments when it's meant to return the sum and reset. You need to check if `num` is `undefined` instead of checking for the type to be 'number'.